scripts/zones/VeLugannon_Palace/mobs/Detector.lua

```diff
@@ -10,25 +10,51 @@ entity.onMobSpawn = function(mob)
     mob:setLocalVar("petCount", 1)
 end
 
+local getMobToSpawn = function(detector)
+    local ID = require("scripts/zones/VeLugannon_Palace/IDs")
+    local detectorID = detector:getID()
+    local canSpawnSteamCleaner = GetServerVariable("[POP]SteamCleaner") < os.time() and utils.contains(detectorID, ID.mob.STEAM_CLEANER_DETECTORS)
+    local steamCleanerSpawnChance = 10 -- percent
+    local steamCleaner = GetMobByID(ID.mob.STEAM_CLEANER)
+
+    if steamCleaner:isSpawned() then
+        -- this check is needed to prevent the detector from spawning BOTH steam cleaner and a caretaker
+        -- by essentially "re-returning" steam cleaner, the subsequent logic will know that he's already spawned as well
+        -- and won't try to respawn him
+        -- This silly check is basically needed because we fall back on a "my id + 1" convention instead of actually
+        -- doing some sort of "this detector owned/spawned this caretaker and while he is alive, I shouldn't summon
+        -- something else".
+        if detector:getLocalVar("SpawnedSC") == 1 then
+            return steamCleaner
+        end
+    elseif canSpawnSteamCleaner and math.random(100) < steamCleanerSpawnChance then
```

Comment:
I don't know if I'm followinf this correctly but...
Wouldn't this prevent ALL detecteors from spawning adds the moment Steam Cleaner is spawned?

Reply:
No, because if it isn't the detector that spawned it, it will fall through to the bottom return. That's whole purpose huge but obvious misleading comment for the if statement on line 27.

Reply:
It's not obvious at a glance, but looking carefully:

The set local var of "SpawnedSC" is assigned to the detector that spawned it, and only that one specific detector will `return SteamCleaner` on subsequent checks. The other detectors will fall through to the bottom return. 

I think adding a comment on the get/set local vars about how they are used on _only_ the detector that spawns it should clear that up for future code spelunkers

Reply:
> It's not obvious at a glance, but looking carefully:
> 
> The set local var of "SpawnedSC" is assigned to the detector that spawned it, and only that one specific detector will `return SteamCleaner` on subsequent checks. The other detectors will fall through to the bottom return.
> 
> I think adding a comment on the get/set local vars about how they are used on _only_ the detector that spawns it should clear that up for future code spelunkers

Let me know if you think these comments are better.